### PR TITLE
Support field-based discoverability control

### PR DESCRIPTION
**Pull request by cheungpat**
connects #377
refs skygeario/features#48

### CODE REVIEW

**Comment by oursky-travis** [comment]
@cheungpat, thanks for your PR! By analyzing the history of the files in this pull request, we identified @rickmak to be a potential reviewer.

**Comment by cheungpat** [comment]
@ben181231 Pull request is updated and is ready for review.

File: pkg/server/handler/query.go [hunk 1]
```diff
@@ -393,3 +394,162 @@ func mapToQueryHookFunc(parser *QueryParser) mapstructure.DecodeHookFunc {
...
+		if c.pStack[len(c.pStack)-1].SimpleComparison {
+			accessMode = skydb.DiscoverOrCompareFieldAccessMode
+		} else {
+			accessMode = skydb.CompareFieldAccessMode
+		}
+	} else if c.inSort {
+		accessMode = skydb.CompareFieldAccessMode
+	} else {
+		accessMode = skydb.ReadFieldAccessMode
+	}
```
Comment (ben181231) [inline pkg/server/handler/query.go]: Need some comments about what do these 3 branches mean.

**Comment by cheungpat** [comment]
@ben181231 updated the PR

**Pull request merged**